### PR TITLE
refactor: return error instead of invoking log.Fatal

### PR DESCRIPTION
**Pull request by honcao**
**Reason for Change**:

Fix issue #963

**Issue Fixed**:

#963

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/armhelpers/azurestack/disk_test.go [hunk 1]
```diff
@@ -9,7 +9,10 @@ import (
 )
 
 func TestDeleteManagedDisk(t *testing.T) {
-	mc := NewHTTPMockClient()
+	mc, err := NewHTTPMockClient()
+	if err != nil {
+		t.Fatalf("failed to create HttpMockClient - %s", err)
```
Comment (jackfrancis) [inline pkg/armhelpers/azurestack/disk_test.go]: Should these be `t.Error()` statements instead of `t.Fatalf()` statements?

Comment (mboersma) [reply pkg/armhelpers/azurestack/disk_test.go]: The testing package's `Fatalf` doesn't involve `os.Exit`, so no worries there:
> Fatalf is equivalent to Logf followed by FailNow.

**Comment by CecileRobertMichon** [comment]
@mboersma feel free to merge this whenever the timing is right for you, I know you have another PR open that might conflict with this one.

**Comment by honcao** [comment]
@CecileRobertMichon  @mboersma , you are welcome.
Thanks for the review and merge.

**Pull request merged**